helix-core/src/main/java/org/apache/helix/task/GenericTaskAssignmentCalculator.java

```diff
@@ -95,6 +97,20 @@
     return placement.computeMapping(jobCfg, jobContext, partitionNums, resourceId);
   }
 
+  @Override
+  public Set<Integer> getRemovedPartitions(JobConfig jobConfig, JobContext jobContext, Set<Integer> allPartitions) {
+    // Get all partitions existed in the context
+    Set<Integer> deletedPartitions = new HashSet<>();
+    // Check whether the tasks have been deleted from jobConfig
+    for (Integer partition : jobContext.getPartitionSet()) {
+      String partitionID = jobContext.getTaskIdForPartition(partition);
+      if (!jobConfig.getTaskConfigMap().containsKey(partitionID)) {
+        deletedPartitions.add(partition);
+      }
+    }
```

Comment:
We can save the loop operation by set operation as:

Set<Integer> deletedPartitions = new HashSet<>(jobContext.getPartitionSet());
deletedPartitions.removeAll(jobConfig.getTaskConfigMap().keySet());

Reply:
I will sync up with you on this offline.

Reply:
Synced offline. Resolved.